client/src/store/reducers/loginReducer.js

```diff
@@ -0,0 +1,29 @@
+const user = JSON.parse(localStorage.getItem("user"));
+
+const defaultState = user
+    ? { isLoggedIn: true, user }
+    : { isLoggedIn: false, user: null };
+
+export const loginReducer = (state = defaultState, action) =>{
+    switch (action.type) {
+        case "logIn":
```

Comment:
Exctract logIn , logOut and other constants variables to actions.js file, check on the Internet redux structure please